### PR TITLE
fix: Add temp_maintenance_status contextmanager

### PR DESCRIPTION
**Pull request by DanielArndt**
This allows us to temporarily set the status of the charm during a long running operation, and then restore it to the previous state once the operation is complete.

Before this change, the "Installing Vault" status message would continue to be set for the remainder of the hook.

# Checklist:

- [ ] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -189,11 +202,11 @@ def _install_vault_snap(self) -> None:
...
-            self.unit.status = MaintenanceStatus("Installing Vault")
-            vault_snap.ensure(
-                snap.SnapState.Latest, channel=VAULT_SNAP_CHANNEL, revision=VAULT_SNAP_REVISION
-            )
-            vault_snap.hold()
+            with self.temp_maintenance_status("Installing Vault"):
+                vault_snap.ensure(
+                    snap.SnapState.Latest, channel=VAULT_SNAP_CHANNEL, revision=VAULT_SNAP_REVISION
+                )
+                vault_snap.hold()
```
Comment (ghislainbourgeois) [inline src/charm.py:209]: This looks awesome, but does it really work? I mean, in juju status, will we see the maintenance status while the snap is installing and then a change to active, all during the same hook?

Comment (gruyaume) [reply src/charm.py:209]: I got surprised by this as well with a different change recently, but yes:

"Changes to status take effect immediately, unlike other Juju operations such as modifying relation data or secrets, which only take effect after a successful event."

Ref: https://ops.readthedocs.io/en/latest/

File: src/charm.py [hunk 2]
```diff
@@ -139,6 +140,18 @@ def __init__(self, *args):
         self.framework.observe(self.on[PEER_RELATION_NAME].relation_created, self._configure)
         self.framework.observe(self.on[PEER_RELATION_NAME].relation_changed, self._configure)
 
+    @contextmanager
+    def temp_maintenance_status(self: CharmBase, message: str):
```
Comment (gruyaume) [inline src/charm.py:144]: Just a nit, but we haven't typed `self` anywhere else than here

Comment (DanielArndt) [reply src/charm.py:144]: yeah it's probably not necessary

Comment (DanielArndt) [reply src/charm.py]: fixed in https://github.com/canonical/vault-operator/pull/80/commits/8a0558733da38eb41b385a7e453b1c6c1eb37be9

**Review by gruyaume — APPROVED** [review]

File: src/charm.py (hunk 1 above)
Comment (DanielArndt) [reply src/charm.py:209]: Yep, tried and tested (manually). I set some different statuses with `sleep` and it works as expected.

**Pull request merged**